lib/CXGN/List/Validate/Plugin/Traits.pm

```diff
@@ -21,26 +21,13 @@ sub validate {
 
     foreach my $term (@$list) {
         print STDERR $term."\n";
-        my $delim = "|";
-        my $full_accession = substr $term, rindex( $term, $delim ) + length($delim);
-        my $full_accession_length = length($full_accession) + length($delim);
-        $term = substr($term, 0, -$full_accession_length);
-        #print STDERR $full_accession."\n";
-        #print STDERR $term."\n";
-        my ($db_name, $accession) = split ":", $full_accession;
-
-        if ($accession) {
-            $accession =~ s/\s+$//;
-            $accession =~ s/^\s+//;
-        }
-        if ($db_name) {
-            $db_name  =~ s/\s+$//;
-            $db_name  =~ s/^\s+//;
-        }
+
+        my ($trait_name, $full_accession) = split (/\|/ , $term);
```

Comment:
This will not validate postcomposed terms properly (since they have | in them)

Reply:
ah, true, I forgot that composed terms now have additional | delimiters. Looks like the other parts of the Fieldbook object didn't account for that either. e2a9ec4a4e91eb4fbd610413a4a54eff44e81858 adds changes to handle the extra delimiters in composed traits. And removes some of the checks in the validation that are no longer needed, now that composed traits are also stored permanently with VARIABLE_OF relationships